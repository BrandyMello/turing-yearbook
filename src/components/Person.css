h1 {
  background-color: rgba(255, 255, 255, 0.95);
  padding: 10px;
  color: limegreen;
}

h2 {
  background-color: rgba(255, 255, 255, 0.95);
  padding: 10px;
  color: #6696EC;
}

.displayCards {
  display: flex;
  align-items: flex-start;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-evenly;
}

img {
  height: 180px;
  width: 60%;
  border-radius: 3%;
}

.card {
  background-color: rgba(255, 255, 255, 0.95);
  border-radius: 3%;
  margin: 10px;
  height: 420px;
  width: 19%;
  padding: 15px;
}

h3 {
  color: #FF6347;
}

.quote {
  font-style: italic;
  font-size: 12px;
  height: 25px;
}

.superlative {
  color: #6696EC;
  font-weight: bold;
  height: 80px;
}

.remove-btn {
  background-color: #ff63479e;
  border-radius: 50%;
  height: 30px;
}