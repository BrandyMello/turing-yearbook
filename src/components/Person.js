import React from 'react';
import './Person.css';

const Person = ({id, photo, name, quote, superlative, addStudent, removeCard}) => {
  return (
      <div className='card'>
        <img src={photo} alt={name}/>
        <h3>{name}</h3>
        <p className='quote' contentEditable="true" suppressContentEditableWarning={true}>"{quote}"</p>
        <p className='superlative' contentEditable="true" suppressContentEditableWarning={true}>{superlative}</p>
        <button className='remove-btn' contentEditable="true" suppressContentEditableWarning={true} onClick={() => removeCard(id)}>Remove</button>
      </div>
    )
}

export default Person;


