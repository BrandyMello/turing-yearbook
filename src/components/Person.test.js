import React from 'react';
import { shallow } from 'enzyme';
import Person from './Person';

describe('Person', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Person
        id={21}
        name='Jane Doe'
        photo='https://placekitten.com/200/300'
        quote='You got this'
        superlative='Most likely to cry'
      addStudent={jest.fn()} 
      removeCard={jest.fn()}
      />);

    expect(wrapper).toMatchSnapshot();
  });
});