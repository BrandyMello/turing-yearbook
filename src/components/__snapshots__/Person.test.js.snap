// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person should match the snapshot 1`] = `
<div
  className="card"
>
  <img
    alt="Jane Doe"
    src="https://placekitten.com/200/300"
  />
  <h3>
    Jane Doe
  </h3>
  <p
    className="quote"
    contentEditable="true"
    suppressContentEditableWarning={true}
  >
    "
    You got this
    "
  </p>
  <p
    className="superlative"
    contentEditable="true"
    suppressContentEditableWarning={true}
  >
    Most likely to cry
  </p>
  <button
    className="remove-btn"
    contentEditable="true"
    onClick={[Function]}
    suppressContentEditableWarning={true}
  >
    Remove
  </button>
</div>
`;
